packages/firestore/src/local/indexeddb_schema.ts

```diff
@@ -793,6 +795,50 @@ export class DbIndexEntries {
   ) {}
 }
 
+export type DbDocumentOverlayKey = [string, string, string];
+
+/**
+ * An object representing a document overlay.
+ */
+export class DbDocumentOverlay {
+  /** Name of the IndexedDb object store. */
+  static store = 'documentOverlays';
+
+  static keyPath = ['userId', 'collectionPath', 'documentId'];
+
+  static batchIdOverlayIndex = 'batchIdOverlayIndex';
+  static batchIdOverlayIndexPath = ['userId', 'largestBatchId'];
```

Comment:
I don't think we need this index. The callsite of `removeOverlaysForBatchId` knows what the actual overlay, so we could just delete by collectionPath + batchId. Most "batches" are of size 1.

Reply:
We should evaluate as part of this PR as this PR changes the schema on disk.

Reply:
I inspected the Android implementation. Indeed the callers of `removeOverlaysForBatchId` have all the document keys for which we want to remove the overlays. I've updated the signature to take a document key set. PTAL.